google/cloud/testing_util/command_line_parsing.cc

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/testing_util/command_line_parsing.h"
+#include "google/cloud/internal/throw_delegate.h"
+#include "absl/strings/match.h"
+#include "absl/time/time.h"
+#include <iomanip>
+#include <sstream>
+
+namespace google {
+namespace cloud {
+inline namespace GOOGLE_CLOUD_CPP_NS {
+namespace testing_util {
+
+// This parser does not validate the input fully, but it is good enough for our
+// purposes.
+std::int64_t ParseSize(std::string const& val) {
+  struct Match {
+    char const* suffix;
+    std::int64_t multiplier;
+  } matches[] = {
+      {"TiB", kTiB}, {"GiB", kGiB}, {"MiB", kMiB}, {"KiB", kKiB},
+      {"TB", kTB},   {"GB", kGB},   {"MB", kMB},   {"KB", kKB},
+  };
+  auto const s = std::stol(val);
+  for (auto const& m : matches) {
+    if (absl::EndsWith(val, m.suffix)) return s * m.multiplier;
+  }
+  return s;
+}
+
+std::size_t ParseBufferSize(std::string const& val) {
+  auto const s = ParseSize(val);
+  if (s < 0 || static_cast<std::uint64_t>(s) >
+                   (std::numeric_limits<std::size_t>::max)()) {
+    internal::ThrowRangeError("invalid range in ParseBufferSize");
+  }
+  return static_cast<std::size_t>(s);
+}
+
+std::chrono::seconds ParseDuration(std::string const& val) {
+  absl::Duration d;
+  if (!absl::ParseDuration(val, &d)) {
+    internal::ThrowInvalidArgument("invalid duration: " + val);
+  }
+  return absl::ToChronoSeconds(d);
+}
+
+absl::optional<bool> ParseBoolean(std::string const& val) {
+  if (val.empty()) {
+    return {};
+  }
```

Comment:
I would suggest omitting this special/redundant case.

Reply:
Done.